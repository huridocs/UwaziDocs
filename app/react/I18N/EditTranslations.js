import React from 'react';

import RouteHandler from 'app/App/RouteHandler';
import { EditTranslationsForm } from 'app/I18N/components/EditTranslationsForm';
import { actions } from 'app/I18N/';
import { I18NApi } from 'app/I18N';

export default class EditTranslations extends RouteHandler {
  static async requestState(requestParams) {
    const translations = await I18NApi.get(requestParams);
    return [actions.editTranslations(translations)];
  }

  render() {
    return (
      <div className="settings-content">
        <EditTranslationsForm context={this.props.params.context} />
      </div>
    );
  }
}
