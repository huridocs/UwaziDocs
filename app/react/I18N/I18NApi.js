import api from 'app/utils/api';

export default {
  get() {
    return api.get('translations')
    .then(response => response.json.rows);
  },

  save(translation) {
    return api.post('translations', translation)
    .then(response => response.json);
  },

  addEntry(context, key, value) {
    return api.post('translations/addentry', { context, key, value })
    .then(response => response.json);
  }
};
