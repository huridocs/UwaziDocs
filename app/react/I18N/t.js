import { store } from 'app/store';
import React from 'react';
import { Translate } from './';

const testingEnvironment = process.env.NODE_ENV === 'test';
const t = (contextId, key, _text, returnComponent = true) => {
  if (returnComponent && !testingEnvironment) {
    return (<Translate context={contextId}>{key}</Translate>);
  }
  const text = _text || key;

  if (!t.translation) {
    const state = store.getState();
    const translations = state.translations.toJS();
    t.translation = translations.find(d => d.locale === state.locale) || { contexts: [] };
  }

  const context = t.translation.contexts.find(ctx => ctx.id === contextId) || { values: {} };

  return context.values[key] || text;
};

t.resetCachedTranslation = () => {
  t.translation = null;
};


export default t;
