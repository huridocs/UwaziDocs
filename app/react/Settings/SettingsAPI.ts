import api from 'app/utils/api';
import { RequestParams } from 'app/utils/RequestParams';
import { Settings } from 'shared/types/settingsType';

export default {
  async save(settings: RequestParams<Settings>): Promise<Settings> {
    const processedSettings = settings.set({
      ...settings.data,
      links: settings.data?.links?.map(link => {
        const { localID, ...linkWithoutLocalID } = link;
        return linkWithoutLocalID;
      }),
    });

    return api.post('settings', processedSettings).then((response: any) => response.json);
  },

  async get(query: RequestParams = new RequestParams()): Promise<Settings> {
    return api.get('settings', query).then((response: any) => response.json);
  },
};
