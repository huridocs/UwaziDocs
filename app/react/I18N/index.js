import I18NLink from './components/I18NLink';
import I18N from './components/I18N';
import I18NUtils from './utils';
import I18NApi from './I18NApi';
import * as actions from './actions/I18NActions';

export { I18NLink, I18NUtils, I18N, I18NApi, actions };
export { I18NMenu } from './components/I18NMenu';
