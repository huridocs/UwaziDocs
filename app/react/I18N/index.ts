import I18NLink from './components/I18NLink';
import I18NUtils from './utils';
import I18NApi from './I18NApi';

export { I18NApi };
export { I18N } from './components/I18N';
export { I18NMenu } from './components/I18NMenu';
export * as actions from './actions/I18NActions';
export { I18NLink as I18NLinkV2 } from './I18NLinkV2';
export { Translate } from './Translate';
export { t } from './translateFunction';
export { TranslateModal } from './TranslateModal';
export { I18NLink, I18NUtils };
