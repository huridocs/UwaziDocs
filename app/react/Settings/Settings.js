import React from 'react';
import Helmet from 'react-helmet';

import RouteHandler from 'app/App/RouteHandler';
import UsersAPI from 'app/Users/UsersAPI';
import ThesaurisAPI from 'app/Thesauris/ThesaurisAPI';
import RelationTypesAPI from 'app/RelationTypes/RelationTypesAPI';
import { actions } from 'app/BasicReducer';
import { I18NApi } from 'app/I18N';
import SettingsNav from './components/SettingsNavigation';
import SettingsAPI from './SettingsAPI';

export class Settings extends RouteHandler {
  static requestState() {
    return Promise.all([
      UsersAPI.currentUser(),
      ThesaurisAPI.getDictionaries(),
      RelationTypesAPI.get(),
      I18NApi.get(),
      SettingsAPI.get()
    ])
    .then(([user, dictionaries, relationTypes, translations, collection]) => (
      { user, dictionaries, relationTypes, translations, settings: { collection } }
    ));
  }

  setReduxState(state) {
    this.context.store.dispatch(actions.set('auth/user', state.user));
    this.context.store.dispatch(actions.set('dictionaries', state.dictionaries));
    this.context.store.dispatch(actions.set('relationTypes', state.relationTypes));
    this.context.store.dispatch(actions.set('translations', state.translations));
    this.context.store.dispatch(actions.set('settings/collection', state.settings.collection));
  }

  render() {
    return (
      <div className="row settings">
        <Helmet title="Settings" />
        <div className="settings-navigation">
          <SettingsNav/>
        </div>
        <div className="settings-content">
          {this.props.children}
        </div>
      </div>
    );
  }
}

export default Settings;
