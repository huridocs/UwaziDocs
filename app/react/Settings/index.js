/* eslint-disable import/no-named-as-default */
import EntityTypesList from './components/EntityTypesList';
import FiltersForm from './components/FiltersForm';
import { RelationTypesList } from './components/RelationTypesList';
import ThesauriList from './components/ThesauriList';
import { Settings } from './Settings';
import SettingsAPI from './SettingsAPI';
import { Dashboard } from './components/Dashboard';
import { PreserveSettings } from './components/PreserveSettings';

export { CustomUploads } from './components/CustomUploads';

export {
  Settings,
  SettingsAPI,
  PreserveSettings,
  EntityTypesList,
  RelationTypesList,
  ThesauriList,
  FiltersForm,
  Dashboard,
};
