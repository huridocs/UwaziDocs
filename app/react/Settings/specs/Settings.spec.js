import UsersAPI from 'app/Users/UsersAPI';
import ThesaurisAPI from 'app/Thesauris/ThesaurisAPI';
import RelationTypesAPI from 'app/RelationTypes/RelationTypesAPI';
import { I18NApi } from 'app/I18N';
import SettingsAPI from '../SettingsAPI';
import { Settings } from '../Settings';

describe('Settings', () => {
  describe('requestState', () => {
    const user = { name: 'doe' };
    const dictionaries = [{ _id: 1, name: 'Countries' }];
    const relationTypes = [{ _id: 1, name: 'Supports' }];
    const translations = [{ _id: 1, locale: 'es', values: {} }];
    const settings = { siteName: 'BatCave' };

    beforeEach(() => {
      spyOn(UsersAPI, 'currentUser').and.returnValue(Promise.resolve(user));
      spyOn(ThesaurisAPI, 'getDictionaries').and.returnValue(Promise.resolve(dictionaries));
      spyOn(RelationTypesAPI, 'get').and.returnValue(Promise.resolve(relationTypes));
      spyOn(I18NApi, 'get').and.returnValue(Promise.resolve(translations));
      spyOn(SettingsAPI, 'get').and.returnValue(Promise.resolve(settings));
    });

    it('should get the current user, and metadata', (done) => {
      Settings.requestState()
      .then((state) => {
        expect(state.user).toEqual(user);
        expect(state.dictionaries).toEqual(dictionaries);
        expect(state.translations).toEqual(translations);
        expect(state.settings.collection).toEqual(settings);
        done();
      });
    });
  });
});
