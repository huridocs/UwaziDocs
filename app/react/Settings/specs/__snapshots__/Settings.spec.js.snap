// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings requestState should get the current user, and metadata 1`] = `
Array [
  Object {
    "type": "auth/user/SET",
    "value": Object {
      "name": "doe",
      "role": "admin",
    },
  },
  Object {
    "type": "dictionaries/SET",
    "value": Array [
      Object {
        "_id": 1,
        "name": "Countries",
      },
    ],
  },
  Object {
    "type": "translations/SET",
    "value": Array [
      Object {
        "_id": 1,
        "locale": "es",
        "values": Object {},
      },
    ],
  },
  Object {
    "type": "settings/collection/SET",
    "value": Object {
      "siteName": "BatCave",
    },
  },
  Object {
    "type": "settings/stats/SET",
    "value": Object {
      "files": 3,
      "users": 2,
    },
  },
]
`;

exports[`Settings requestState should not request the stats if the user does not have admin role 1`] = `
Array [
  Object {
    "type": "auth/user/SET",
    "value": Object {
      "name": "jane",
      "role": "editor",
    },
  },
  Object {
    "type": "dictionaries/SET",
    "value": Array [
      Object {
        "_id": 1,
        "name": "Countries",
      },
    ],
  },
  Object {
    "type": "translations/SET",
    "value": Array [
      Object {
        "_id": 1,
        "locale": "es",
        "values": Object {},
      },
    ],
  },
  Object {
    "type": "settings/collection/SET",
    "value": Object {
      "siteName": "BatCave",
    },
  },
  Object {
    "type": "settings/stats/SET",
    "value": Object {},
  },
]
`;
